client/Contexts/Introspect/introspection.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import {IntrospectionSchema, IntrospectionType, IntrospectionTypeRef,} from 'graphql';
+import {Entities, Entity, Expectations, OperationCallback} from "./ExpectationTypes";
+
+/**
+ * Transforms a field from an object representation to a GraphQL schema string representation
+ * @param type the type to transform
+ * @return the GraphQL schema string
+ */
+const transformFieldType = (type: IntrospectionTypeRef) => {
+    if (type.kind === 'LIST') {
+        return `[${transformFieldType(type.ofType)}]`;
+    } else if (type.kind === 'NON_NULL') {
+        return `${transformFieldType(type.ofType)}!`;
+    } else if (
+        type.kind === 'SCALAR' ||
+        type.kind === 'OBJECT' ||
+        type.kind === 'ENUM'
+    ) {
+        return type.name;
+    } else {
+        throw new Error(`Unsupported graphql kind ${type.kind} for ${type.name}`);
+    }
+};
+
+/**
+ * Converts an array into an object, using the element of the array indexed with `by` as the key.
+ * @param array The array to convert
+ * @param by The index of the array to use as the key
+ * @return the object
+ */
+const indexBy = <T>(array: readonly T[], by: string) => {
+    return array.reduce((acc, curr) => {
```

Comment:
Array of `T`, reduce into a single object hash whereby each `T` is stored using its value of `by` as the key in the hash - what if value of `by` is the same/missing for multiple items? And does TS handle cases where the value isn't a string?

Also I think referring to `index` is confusing - it's not the index of the array, it's the prop name to use as the key.

Reply:
> Array of T, reduce into a single object hash whereby each T is stored using its value of by as the key in the hash - what if value of by is the same/missing for multiple items?

I don't think it can be, in the usage of this function here (as the introspection schema won't return the same thing twice), but I'll add an explicit check and document it.

> And does TS handle cases where the value isn't a string?

I don't think it does so I've added an explicit check and documented it. Again, I don't believe in the usage of the function this error can occur (due to what the introspection schema returns).

> Also I think referring to index is confusing - it's not the index of the array, it's the prop name to use as the key.

Used your text.